### DISCUSSION TITLE
Daily Journal Template (or View)

### DISCUSSION THREAD

**Discussion by stdword** [stdword/logseq13-full-house-plugin#6] [Showroom]
<img width="600px" src="https://user-images.githubusercontent.com/1984175/229267622-05da1ff3-1b5c-4632-a59c-fa0dc951e8a1.gif"/>

## Steps

1) Setup standard Logseq template (eg with name "my day")
2) Configure Logseq to use it in `config.edn`:
    ```clojure
    :default-templates {:journals "my day"}
    ```
3) Create template with everything you need. Let's get this one with days navigation (use copy button in the top right corner):
    ```markdown
    - template:: journal-embed
      template-version:: 3
      plugin-version:: 3.0.0
      link:: https://github.com/stdword/logseq13-full-house-plugin/discussions/6
      - ``{ if (!c.page.day) return '[not in a journal page]' }``
        ← [Yesterday](``[c.page.day.subtract(1, 'day')]``) | [Tomorrow](``[c.page.day.add(1, 'day')]``) →
    ```
4) Add rendering code to "my day" template:
    - to use it as `🏛view`: 
      ```edn
      {{renderer :template-view, journal-embed}}
      ```
    - to use it as `🏛template`:
      ```edn
       {{renderer :template, journal-embed, :delay-until-rendered}}
       ```

<details closed><summary><b>Steps in video</b></summary>
  <video width="40%" src="https://user-images.githubusercontent.com/1984175/229267085-8819a19a-ae15-4d01-a63b-069f1a1649f3.mp4"/>
</details>

## Additional info
- Logseq v0.9.20
- ☕️ [BuJo Theme](https://github.com/stdword/logseq-bujo-theme)
- Plugins: [Banners](https://github.com/yoyurec/logseq-banners-plugin) v3.4.10

**Comment by ard0gg** [comment]
This seems to be broken now. Is there a better way to write this with updated syntax?

**Reply by ard0gg** [reply]
This still works. I hadn't used logseq in a while. It appears I need "Show brackets" enabled in general settings or links have line breaks before and after? Could be the results of some plugin I had enabled to.

**Reply by ard0gg** [reply]
FWIW, I did need to modernize this a bit for my template. This is what i came up with 

```` js
- template:: journal-embed
  template-type:: full house
  template-version:: 2.0
  plugin-version:: 4.1.0
  link:: [Daily Journal Template (or View) · stdword/logseq13-full-house-plugin · Discussion #6 · GitHub](https://github.com/stdword/logseq13-full-house-plugin/discussions/6)
	- ```javascript
	  ``{           
	      
	      if (!c.page.day) {
	           return '[Not a date Page]'
	      }
	      _ 
	  }``
	    ← [Previous Day](``@yesterday, page``) | [Today](``@today, page``) | [Next Day](``@tomorrow, page``) → 
	    [[project]] | [[area]] | [[resource]] | [[archive]]
	  ```
````

PS I have no idea which language to reference for syntax highlighting.

**Reply by stdword** [reply]
Hello! It works for me without any modifications:
<img width="300px" src="https://github.com/user-attachments/assets/66fffc1e-6f76-4ed8-9c07-2be8a1c013c6"/>

But there is really an issue with  «Show brackets» turned off:
<img width="180px" src="https://github.com/user-attachments/assets/b43a2088-9370-4bc2-8abb-157a4593f834"/>

I need some time to figure out what's going on

**Reply by stdword** [reply]
@ard0gg, fixed in v4.3.6

**Reply by ard0gg** [reply]
Thanks for this quick reply and fix! Confirmed working on my end as well.
